kDrive/UI/Controller/Files/Categories/ManageCategoriesViewController.swift

```diff
@@ -252,8 +252,14 @@ final class ManageCategoriesViewController: UITableViewController {
             return
         }
         self.driveFileManager = driveFileManager
-        let realm = driveFileManager.getRealm()
-        files = filesId.compactMap { driveFileManager.getCachedFile(id: $0, using: realm) }
+
+        let matchedFiles = driveFileManager.fetchResults(ofType: File.self) { realm in
+            realm.objects(File.self)
+                .filter("id IN %@", filesId)
```

Comment:
I migrated to a NSPredicate to be able to use a read transaction

Reply:
Maybe we want to use the strongly typed         
```suggestion
                .where {
                    $0.id.in(filesId)
                }
```

Reply:
Well I'm not sure, it depends what we want to do abstraction wise 

https://github.com/Infomaniak/ios-kDrive/pull/1167#issuecomment-2058623898